### PR TITLE
adds Gaussian gate

### PR DESCRIPTION
**Pull request by ziofil**
Ggate for lab_dev

### CODE REVIEW

File: mrmustard/lab_dev/transformations/ggate.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+class Ggate(Unitary):
+    r"""
+    The generic N-mode Gaussian gate.
+
+    .. code-block ::
+
+        >>> from mrmustard import math
+        >>> from mrmustard.lab_dev import Ggate, Vacuum, Identity
```
Comment (ziofil) [inline mrmustard/lab_dev/transformations/ggate.py:38]:
```suggestion
        >>> from mrmustard.lab_dev import Ggate, Vacuum, Identity, Ket
```

**Review by arsalan-motamedi — APPROVED** [review]

File: mrmustard/lab_dev/transformations/base.py [hunk 2]
```diff
@@ -191,14 +191,12 @@ def from_symplectic(cls, modes, S) -> Unitary:
         modes: the modes that we want the unitary to act on (should be a list of int)
         S: the symplectic representation (in XXPP order)
         """
-        m = len(S)
+        m = len(modes)
         A = symplectic2Au(S)
-        b = math.zeros(m, dtype="complex128")
+        b = math.zeros(2 * m, dtype="complex128")
         c = complex(1)  # TODO: to be change after poly*exp ansatz
```
Comment (apchytr) [inline mrmustard/lab_dev/transformations/base.py:197]:
```suggestion
```

File: tests/test_lab_dev/test_transformations/test_ggate.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+"""
+Test for the Ggate class.
+"""
+
+from mrmustard.lab_dev import Ggate, Identity
+from mrmustard import math
+
+
+def test_ggate_is_unitary_1mode():
```
Comment (apchytr) [inline tests/test_lab_dev/test_transformations/test_ggate.py:23]: Can we change this to match our other unit tests? As in, 
```
TestGgate:
r"""
"""

def test_init()
...

def test_ggate_is_unitary_1mode()
...
```
You can you the other test files as an example

Comment (arsalan-motamedi) [reply tests/test_lab_dev/test_transformations/test_ggate.py:23]: Sure, will do.

**Review by apchytr — APPROVED** [review]

File: mrmustard/lab_dev/circuit_components.py [hunk 4]
```diff
@@ -373,10 +373,7 @@ def quadrature(self, quad: Batch[Vector], phi: float = 0.0) -> ComplexTensor:
         if isinstance(self.representation, Fock):
             fock_arrays = self.representation.array
             # Find where all the bras and kets are so they can be conjugated appropriately
-            conjugates = [
-                False if i in self.wires.ket.indices else True
-                for i in range(len(self.wires.indices))
-            ]
+            conjugates = [i not in self.wires.key.indices for i in range(len(self.wires.indices))]
```
Comment (apchytr) [inline mrmustard/lab_dev/circuit_components.py:376]:
```suggestion
            conjugates = [i not in self.wires.ket.indices for i in range(len(self.wires.indices))]
```

**Review by apchytr — APPROVED** [review]

**Review by elib20 — APPROVED** [review]

**Pull request merged**